src/compiler/crystal/compiler.cr

```diff
@@ -546,6 +562,15 @@ module Crystal
         bc_name = self.bc_name
         object_name = self.object_name
 
+        {% unless LibLLVM::IS_38 || LibLLVM::IS_39 %}
+        if compiler.thin_lto
+          @llvm_mod.write_bitcode_with_summary_to_file(object_name)
+          @reused_previous_compilation = false
+          llvm_mod.print_to_file ll_name if compiler.dump_ll?
+          return
```

Comment:
I've been wondering why we dumped the LLVM IR in this block, then understood this is because we are skipping the rest of the method. Maybe extract methods to keep a normal flow, instead of adding a duplication?

```crystal
if compiler.thin_lto
  emit_bc_with_thin_lto_summary
else
  emit_bc_and_compile_object
end

dump_llvm_ir
```

Reply:
@ysbaddaden I updated it it something like that. The (private) `compile_to_thin_lto` is emitted always, but raises if the llvm version does not support lto. 

Yet, the invocation will never occur since the option is never toggled to `true` in an non lto llvm version.